trustfall_core/src/interpreter/hints/tests/mod.rs

```diff
@@ -580,6 +580,31 @@ mod static_property_values {
         assert_eq!(adapter.on_starting_vertices.borrow()[&vid(1)].calls, 1);
     }
 
+    #[test]
+    fn required_properties_test() {
+        let input_name = "required_properties";
+
+        let adapter = TestAdapter {
+            on_starting_vertices: btreemap! {
+                vid(1) => TrackCalls::<ResolveInfoFn>::new_underlying(Box::new(|info| {
+                    assert!(info.coerced_to_type().is_none());
+                    assert_eq!(vid(1), info.vid());
+
+                    assert_eq!(vec![RequiredProperty::new("value".into()),
+                                    RequiredProperty::new("multiple".into()),
```

Comment:
I think it makes sense to return `multiple` here because we are folding it, but if we were not using `@fold` I think we should exclude it, correct @obi1kenobi ? I need to confirm it works as intended and fix it.

Reply:
`multiple` is an edge, not a property, so it shouldn't be returned.

Only `__typename, name, value` are properties used at Vid 1 in the test query. `predecessor` and `multiple` are edges, not properties.

Properties are fields whose type is scalar or list of scalar (or list of list of scalar etc.). You can't use a `{` next to a property, there's nothing to expand into. Edges are fields whose type is an interface or object type, or list of such, and they are almost always used with `{` next to them.

Reply:
Thanks! That totally makes sense.